container_main.py

```
@@ -66,19 +67,39 @@ def run_scheduler():
+        port_env = os.environ.get("PORT", "5000")
+        if not port_env.isdigit():
+            logger.error(f"PORT must be a number, got: {port_env}")
+            sys.exit(1)
+
+        # Use gunicorn for production instead of Flask development server
+        # Command is strictly constructed to prevent injection
+        port_safe = shlex.quote(port_env)
+        bind_address = f"0.0.0.0:{port_safe}"
+
+        logger.info(f"Starting Gunicorn on {bind_address}")
+        subprocess.run(
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)